Shell/Skolem.cpp

```diff
@@ -393,23 +400,55 @@ Formula* Skolem::skolemise (Formula* f)
       VarSet::Iterator vuIt(*dep);
       while(vuIt.hasNext()) {
         unsigned uvar = vuIt.next();
-        domainSorts.push(_varSorts.get(uvar, Sorts::SRT_DEFAULT));
-        fnArgs.push(TermList(uvar, false));
-        Formula::VarList::push(uvar,var_args);
+        TermList sort = _varSorts.get(uvar, Term::defaultSort());
+        if(sort == Term::superSort()){
+          args.push(TermList(uvar, false));
+          vArgs.pushFront(uvar);
+        } else {
+          if(sort.isVar() || !sort.term()->shared() || !sort.term()->ground()){
+            sort = SubstHelper::apply(sort, _subst);
+          }
+          argSorts.push(sort);
+          termArgs.push(TermList(uvar, false));
+          vArgs.pushBack(uvar);
+        }
         arity++;
       }
+      ASS(termArgs.size() == argSorts.size());
+
+      VList* vl = VList::empty();
+      for(int i = args.size() -1; i >= 0 ; i--){
+        VList::push(args[i].var(), vl);
+      }
 
-      Formula::VarList::Iterator vs(f->vars());
+      for(unsigned i = 0; i < termArgs.size() && !_appify; i++){
```

Comment:
Wow, isn't this simply the same as 

```
if (_appify) {
  for ...
```

?

Reply:
It is, I prefer the `&& !_appify` because of its conciseness, but I can imagine that maybe it is a bit confusing.

Reply:
I won't say you should fix it. But would vote against using such idioms for code that you expect others to read. 

Sometimes _concise_ is not the same thing as _easy to digest while skimming through_.